### PR TITLE
[name] Implement approximate language matching

### PR DESCRIPTION
**Pull request by behdad**
Very rudimentary.

Fixes https://github.com/harfbuzz/harfbuzz/issues/3354

### CODE REVIEW

File: src/hb-ot-name-table.hh [hunk 1]
```diff
@@ -169,16 +169,31 @@ _hb_ot_name_entry_cmp_key (const void *pa, const void *pb)
...
+
+  signed c = strcmp (astr, bstr);
+
+  if (!exact && c)
+  {
+    unsigned la = strlen (astr);
+    unsigned lb = strlen (bstr);
+    // 'a' is the user request, and 'b' is string in the font.
+    // If eg. user asks for "en-us" and font has "en", approve.
+    if (la > lb && astr[lb] == '-')
```
Comment (matthiasclasen) [inline src/hb-ot-name-table.hh:184]: You don't actually check that the prefixes match here ? Needs a strncmp (astr, bstr, lb) == 0

It doesn't matter for these short strings, but you could write that check without entirely iterating both strings twice,

File: test/api/test-ot-name.c [hunk 2]
```diff
@@ -93,10 +93,15 @@ test_ot_name (void)
   name_id = entries[3].name_id;
   g_assert_cmpuint (3, ==, name_id);
   lang = entries[3].language;
+
   g_assert_cmpstr (hb_language_to_string (lang), ==, "en");
   g_assert_cmpuint (27, ==, hb_ot_name_get_utf8 (face, name_id, lang, &text_size, text));
   g_assert_cmpuint (9, ==, text_size);
   g_assert_cmpstr (text, ==, "FontForge");
+
+  g_assert_cmpuint (27, ==, hb_ot_name_get_utf8 (face, name_id, hb_language_from_string ("en_US", -1), &text_size, text));
```
Comment (matthiasclasen) [inline test/api/test-ot-name.c:102]: your fuzzy check above looks for '-', this uses '_'. Does hb_language_from_string normalize ?

Comment (behdad) [reply test/api/test-ot-name.c:102]: It does normalize.

File: src/hb-ot-name-table.hh (hunk 1 above)
Comment (behdad) [reply src/hb-ot-name-table.hh]: Oops right. Fixed using the slow strncmp version.

**Review by matthiasclasen — COMMENTED** [review]
Looks good now.

**Pull request merged**